crates/iceberg-catalog/src/api/management/v1/warehouse.rs

```diff
@@ -159,8 +162,36 @@ impl axum::response::IntoResponse for CreateWarehouseResponse {
 impl<C: Catalog, A: AuthZHandler, S: SecretStore> Service<C, A, S> for ApiServer<C, A, S> {}
 
 #[async_trait::async_trait]
-
 pub trait Service<C: Catalog, A: AuthZHandler, S: SecretStore> {
+    async fn create_user(
+        context: ApiContext<State<A, C, S>>,
+        request_metadata: RequestMetadata,
+    ) -> Result<User> {
+        let auth = request_metadata
+            .auth_details
+            .ok_or(ErrorModel::bad_request(
+                "Auth details found in request metadata",
+                "MissingUserId",
+                None,
+            ))?;
+        let user_id = auth.user_id().ok_or(ErrorModel::bad_request(
+            "User ID not found in auth details",
+            "MissingUserId",
+            None,
+        ))?;
+
+        let user_id = Uuid::new_v5(&user_id, auth.issuer().as_bytes());
```

Comment:
This is future proofing to avoid clashing user_ids across idps. Not sure if necessary, it also obfuscates lookups by user_id in our db. Should probably kick this in favor of a new idp table which we'll add once we go to multiple idps, then the unique constraint could become `(user_id, idp_id)` which would make the problem go away.